### DISCUSSION TITLE
Unable to resolve peer dependency

### DISCUSSION THREAD

**Discussion by nursahketene** [vercel/next.js#25398] [Help]
### What version of Next.js are you using?

10.2.0

### What version of Node.js are you using?

15.8.0

### What browser are you using?

Firefox, Safari

### What operating system are you using?

macOs

### How are you deploying your application?

N/A

### Describe the Bug

installing packages returns an "Unresolved peer dependency" error.

```npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   peer react@"^16.9.0 || ^17" from @next/react-dev-overlay@10.2.0
npm ERR!   node_modules/@next/react-dev-overlay
npm ERR!     @next/react-dev-overlay@"10.2.0" from next@10.2.0
npm ERR!     node_modules/next
npm ERR!       next@"^10.2.0" from the root project
npm ERR!   peer react@"^16.6.0 || ^17" from next@10.2.0
npm ERR!   node_modules/next
npm ERR!     next@"^10.2.0" from the root project
npm ERR!   25 more (react-dom, styled-jsx, use-subscription, ...)
```

### Expected Behavior

I would expect the package installed with successful outcome. 

### To Reproduce

`npm install --save-dev jest`

**Comment by nursahketene** [comment]
The peerDependencies should be changed according to the requirements of the npm packaging.

From: 
```
"peerDependencies": {
  "react": "^16.0.0"
}
```

To

```
"peerDependencies": {
  "react": "~16.8.6 || ~16.9.0 || ~16.11.0 || ~16.13.1 || ~17.0.2",
},
```


Here is the related [article](https://github.blog/2021-02-02-npm-7-is-now-generally-available/#peer-dependencies)

**Comment by timneutkens** [comment]
The current peerdependencies are correct, not sure where you got the peerDependencies in the comment from but Next.js has `"react": "^16.9.0 || ^17"` for all deps that rely on React

**Reply by nursahketene** [reply]
I have received it from  @next/react-dev-overlay@10.2.0.

**Reply by timneutkens** [reply]
https://unpkg.com/@next/react-dev-overlay@10.2.0/package.json

```
  "peerDependencies": {
    "react": "^16.9.0 || ^17",
    "react-dom": "^16.9.0 || ^17"
  },
```

**Reply by nursahketene** [reply]
Hmm. I will check it again where the error actually happens by going through the error messages
